substrate/frame/contracts/src/lib.rs

```diff
@@ -656,7 +656,63 @@ pub mod pallet {
 				"Debug buffer should have minimum size of {} (current setting is {})",
 				MIN_DEBUG_BUF_SIZE,
 				T::MaxDebugBufferLen::get(),
-			)
+			);
+
+			// Validators have configured runtime memory for 512MB. They need to keep in it
+			// max_runtime_mem, PoV in multiple copies (1x encoded + 2x decoded), and storage
+			// including events. The assumption is that max_runtime_mem + storage/events can be
+			// a maximum of half of the validator runtime memory.
+			let max_block_ref_time = T::BlockWeights::get()
+				.get(DispatchClass::Normal)
+				.max_total
+				.unwrap_or_else(|| T::BlockWeights::get().max_block)
+				.ref_time();
+			let max_payload_size = T::Schedule::get().limits.payload_len;
+			// Blake2_128Concat is being used.
+			let max_key_size = T::MaxStorageKeyLen::get().saturating_add(128);
```

Comment:
isn't it better here to create a key of MaxStorageKeyLen, hash it and get it's len rather than assuming blake2_128Concat being used

Reply:
Yeah I think that would be good. Construct a `Key::Var` and call `hash` on it would be more robust against code changes.

Reply:
Done